libr/core/cmd_anal.c

```diff
@@ -855,16 +855,20 @@ static void __add_vars_sdb(RCore *core, RAnalFunction *fcn) {
 
 	r_list_foreach (all_vars, iter, var) {
 		if (var->isarg) {
-			char *query = r_str_newf ("anal/types/func.%s.arg.%d=%s,%s", fcn->name, arg_count, var->type, var->name);
-			sdb_querys (core->sdb, NULL, 0, query);
-			free (query);
+			char *key = r_str_newf ("func.%s.arg.%d", fcn->name, arg_count);
+			char *value = r_str_newf ("%s,%s", var->type, var->name);
+			sdb_set (core->anal->sdb_types, key, value, 0);
+			free (key);
+			free (value);
 			arg_count++;
 		}
 	}
 	if (arg_count > 0) {
-		char *query = r_str_newf ("anal/types/func.%s.args=%d", fcn->name, arg_count);
-		sdb_querys (core->sdb, NULL, 0, query);
-		free (query);
+		char *key = r_str_newf ("func.%s.args", fcn->name);
+		char *value = r_str_newf ("%d", arg_count);
+		sdb_set (core->anal->sdb_types, key, value, 0);
```

Comment:
what about using sdb_num_set and avoid the strnewf/free of the value?

Reply:
Tried it before. That sets it in hexadecimal form and breaks some tests.

Reply:
The numbers in sdb should be valid in any base. so if that fails it means that somewhere else is not using the api and should be fixed. can you try to find out the place?

Reply:
There are quite a few of those. From what I understood in telegram discussion goal of Analysis project and changes done by @thestr4ng3r is to get rid of direct sdb access for such operations and maybe even sdb. I don't think it's worth refactoring within this issue knowing that such code will be removed. This PR supposed to be a small change to avoid a large performance issue without changing behavior.